force package-lock.json to reference https

ChrisBAshton opened the pull request:
This is currently the advised thing to do while NPM fixes the bug: https://npm.community/t/npm-install-downgrading-resolved-packages-from-https-to-http-registry-in-package-lock-json/1818/3

Test notes: there should be no `http:` references in any `package-lock.json` file in this repository, after running an `npm install`.

- [x] I have assigned myself to this PR and the corresponding issues
- [ ] Tests added for new features
- [x] Test engineer approval

ChrisBAshton committed:
force package-lock.json to reference https

This is currently the advised thing to do while NPM fixes the bug: https://npm.community/t/npm-install-downgrading-resolved-packages-from-https-to-http-registry-in-package-lock-json/1818/3

sareh reviewed (dismissed):
👍 Looks good

Commit:
fix merge conflict

Commit:
update http to https for package after merge conflict

pjlee11 commented:
[Jenkins failure](https://ci.news.tools.bbc.co.uk/blue/organizations/jenkins/psammead/detail/force-https-package-lock/4/pipeline) - when running `make install` on CI the following error is thrown:

```
> @bbc/psammead@0.1.0 force-package-lock-https /var/lib/jenkins/workspace/ad_force-https-package-lock-XOD5QFKGO7J6HJG5KVAKMLMPJNORCJAULJKDNZQLHT4D6EPR5WYQ@4

> find . -type f -name "package-lock.json" -print0 | xargs -0 sed -i '' -e 's/http:\/\//https:\/\//g'

sed: can't read : No such file or directory
```

My guess is that the difference between `npm install` locally and `npm ci` on Jenkins is causing a valid error

jamesbrumpton committed:
Merge branch 'latest' into force-https-package-lock

sareh committed:
Merge branch 'latest' into force-https-package-lock

ChrisBAshton committed:
only run https-fixer locally, not on Jenkins

We only run the https fixer so that we can have consistent package-lock.json files in our commits. Jenkins never runs commits so we don't have to worry

ChrisBAshton committed:
merge latest

pjlee11 reviewed (changes requested):
Jenkins failing again, now on `sh: 1: [[: not found` following the addition of `if ! [[ -n $JENKINS_HOME ]]; then \n npm run force-package-lock-https \n fi`

ChrisBAshton committed:
let's figure out what we're working with

ChrisBAshton committed:
use safer bash command

[[ is a bash built-in, which is not available on Jenkins

bcmn reviewed (dismissed):
Looks good & works locally. 👍  I also didn't know what `test -z` did until just now -- handy. 👀

pjlee11 reviewed (dismissed):
Nice solution!

![](https://media.giphy.com/media/OE6FE4GZF78nm/giphy.gif)

Commit:
Merge branch 'latest' into force-https-package-lock

ChrisBAshton committed:
make command execution more obvious

ChrisBAshton committed:
move http->https to precommit hook

package-lock.json is created AFTER the postinstall step is run, so it's pointless running the http->https step until later. Best place to do this in the precommit hook, which is never run on Jenkins so we don't need the CI check any more

ChrisBAshton committed:
Merge branch 'force-https-package-lock' of https://github.com/BBC-News/psammead into force-https-package-lock

ChrisBAshton committed:
run force-package-lock-https in postshrinkwrap task

the git commit https://github.com/BBC-News/psammead/pull/195/commits/d15748c9f110ff0ce573d4cb3af6b272f325f371 approach doesn't work as changing from http -> https precommit means any changed package-lock files need re-staging and re-committing! So the precommit step is too late

bcmn commented:
![screenshot 2018-12-14 at 11 12 09](https://user-images.githubusercontent.com/11646957/50000316-1e63d500-ff91-11e8-9d49-7a1214b0b44e.png)

The logging shows that this is altering node_modules? I can't imagine this'll ever be an issue, but it feels weird.

ChrisBAshton commented:
argh you're right. Needs a tighter fix...

ChrisBAshton committed:
ignore node_modules in the http->https replacement

ChrisBAshton commented:
Fixed!

bcmn committed:
Merge branch 'latest' into force-https-package-lock

dr3 commented on package.json:
```diff
@@ -8,8 +8,10 @@
+    "force-package-lock-https": "find . -type f -name \"package-lock.json\" | grep -v node_modules | xargs -I % sh -c 'echo \"Replacing http->https in %\"; sed -i \"\" -e \"s/http:\\/\\//https:\\/\\//g\" %'",
     "install:packages": "lerna exec -- npm install",
     "postinstall": "npm run ci:packages",
+    "postshrinkwrap": "if test -z $JENKINS_HOME; then \n npm run force-package-lock-https \n fi",
```
```suggestion
    "postshrinkwrap": "if test -z $CI; then \n npm run force-package-lock-https \n fi",
```
Uses https://github.com/BBC-News/psammead/blob/latest/Jenkinsfile#L22

Commit:
use $CI instead of $JENKINS_HOME

Co-Authored-By: ChrisBAshton <ChrisBAshton@users.noreply.github.com>

sareh reviewed (approved):
👍

bcmn committed:
Merge branch 'latest' into force-https-package-lock

jamesbrumpton commented:
LGTM
